### PR TITLE
libreoffice: fix *.desktop files Exec commands

### LINKED ISSUE #196981: LibreOffice 7.3.6.2 & 7.4.2.4 *.desktop files have incorrect binary names in Exec command
### Describe the bug

The current unstable versions of libreoffice-still and libreoffice-fresh are unable to be launched via their *.desktop files as found in the derivation's `share/applications/` directory.

This means that for me, Rofi can no longer launch LibreOffice Writer etc.

### Steps To Reproduce
1. Update unstable channel
2. Add `libreoffice`, `libreoffice-still` or `libreoffice-fresh` as a package to be installed
3. nixos-rebuild switch
4. Try and run Writer or any other LibreOffice app from a desktop launcher

### Expected behavior

Should be able launch LibreOffice applications from a desktop launcher, e.g. Rofi.

### Screenshots

2022-10-20-22-17-47_3000x2000_scrot

### Additional context

The desktop files are expecting `libreoffice7.3` as the binary name, but it's `libreoffice`:

2022-10-20-22-24-15_2574x820_scrot

It looks like in https://github.com/NixOS/nixpkgs/pull/196535 the renaming of binaries in *.desktop was removed.

The following was taken out for some reason...

```bash
    for f in $out/share/applications/*.desktop; do
      substituteInPlace "$f" \
        --replace "Exec=libreofficedev${major}.${minor}" "Exec=libreoffice" \
        --replace "Exec=libreoffice${major}.${minor}"    "Exec=libreoffice"
    done
```

### Notify maintainers

@gador

### Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.0.2, NixOS, 22.11 (Raccoon), 22.11pre418904.db25c4da285`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(root): `"nixos, nixos2205-22.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

### PR DESCRIPTION
**Pull request by ianmjones**
###### Description of changes

Resolves #196981

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gador** [comment]
Did you test it? I took it out, since the `Exec` line changed to a relative call..

I'll look into it later

**Comment by bobby285271** [comment]
> since the Exec line changed to a relative call..

Can you provide a link to the related commits (files)? I though the commit message in https://github.com/NixOS/nixpkgs/commit/64661f0597a49e9836d3b9cb5c1bdea5e7624922 still applies :thinking: 

https://github.com/LibreOffice/core/blob/libreoffice-7-4/instsetoo_native/util/openoffice.lst.in
https://github.com/LibreOffice/core/blob/libreoffice-7-4/sysui/desktop/menus/base.desktop

**Comment by gador** [comment]
`Exec=${UNIXBASISROOTNAME} --base %%FILE%%`

This should resolve to `libreoffice -- base %FILE%`

The `suvstituteInPlace` failed after the update, this is how I noticed.

Also the desktop files of this update work without issues on my i3 system.

Ill look into it later. I don't have my laptop on me right now

**Comment by ianmjones** [comment]
@gador said...

> Did you test it?

Of course! :smile: 

> I took it out, since the Exec line changed to a relative call

If you look at the second screenshot in #196981 you'll see that I grep the `Exec` line in the desktop file for Writer, it has `libreoffice7.3` as the binary name, which does not exist in the `bin` directory.

> Also the desktop files of this update work without issues on my i3 system.

I'm using i3 too, without this fix, I can't launch LibreOffice apps via rofi.

> Obviously sometimes it isn't libreoffice but the -dev version.

Does putting back the code in this fix not cover that already?

**Comment by gador** [comment]
For some reason, libreoffice starts just fine with dmenu on my system..

I mistook the `UNIXBASISROOTNAME` for `libreoffice` and not `linbreoffice{major}.{minor}` and was thrown off the `substituteInPlace` call failing. So your right and this PR should be merged. I just have a small change: AFAIK `libreofficedev` isn't used anymore and should be removed. (This was probably what threw the errors anyway)

File: pkgs/applications/office/libreoffice/default.nix [hunk 1]
```diff
@@ -364,6 +364,12 @@ in
     ln -s $out/bin/soffice $out/bin/libreoffice
     ln -s $out/lib/libreoffice/share/xdg $out/share/applications
 
+    for f in $out/share/applications/*.desktop; do
+      substituteInPlace "$f" \
+        --replace "Exec=libreofficedev${major}.${minor}" "Exec=libreoffice" \
```
Comment (gador) [inline pkgs/applications/office/libreoffice/default.nix:369]:
```suggestion
```

**Review by gador — CHANGES_REQUESTED** [review]

**Comment by 7c6f434c** [comment]
Should it `default.nix` or in the wrapper?

**Comment by gador** [comment]
> Should it `default.nix` or in the wrapper?

I would suggest leaving it `default.nix` as it belongs to the package itself (and not necessarily the wrapper)

**Comment by 7c6f434c** [comment]
>I would suggest leaving it `default.nix` as it belongs to the package itself (and not necessarily the wrapper)

It's about superficial additions just to launch the code in various environments, and seems prone to future breakage if the naming scheme drifts, that would be nice to fix without rebuilds for testing…

**Comment by gador** [comment]
> >I would suggest leaving it `default.nix` as it belongs to the package itself (and not necessarily the wrapper)
> 
> It's about superficial additions just to launch the code in various environments, and seems prone to future breakage if the naming scheme drifts, that would be nice to fix without rebuilds for testing…
> 

That's a good point. I haven't thought of that. Changes in `.desktop` file have been rare upstream so I wouldn't expect a name drift soon. 
Maybe we should keep that idea for another PR and first merge this one to fix the existing issue of the desktop files not working?

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 196994` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
12 packages built:
</summary>
<ul>
<li>jpsxdec</li>
<li>libreoffice (libreoffice-still)</li>
<li>libreoffice-fresh</li>
<li>libreoffice-fresh-unwrapped</li>
<li>libreoffice-qt</li>
<li>libreoffice-unwrapped (libreoffice-still-unwrapped)</li>
<li>paperwork</li>
<li>python310Packages.paperwork-backend</li>
<li>python310Packages.paperwork-shell</li>
<li>python39Packages.paperwork-backend</li>
<li>python39Packages.paperwork-shell</li>
<li>unoconv</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
libreoffice-fresh:
<details>

got build log for '/nix/store/sr0bag0iy0dqdja404gkdvwi7yir84vs-libreoffice-7.4.2.3' from 'daemon'
When evaluating attribute ‘libreoffice-fresh’:
warning: unused-argument
Unused argument: `commonsLogging`.
Near pkgs/applications/office/libreoffice/default.nix:89:3:

```
   |
89 | , commonsLogging
   |   ^
```
</details>
libreoffice-fresh-unwrapped:
<details>

got build log for '/nix/store/d1s2babi48pf8iwp2xgc28samzibnc93-libreoffice-7.4.2.3' from 'daemon'
When evaluating attribute ‘libreoffice-fresh-unwrapped’:
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'Exec=libreofficedev7.4' doesn't match anything in file '/nix/store/d1s2babi48pf8iwp2xgc28samzibnc93-libreoffice-7.4.2.3/share/applications/base.desktop'
Near pkgs/applications/office/libreoffice/default.nix:578:

```
    |
578 |     description = "Comprehensive, professional-quality productivity suite, a variant of openoffice.org";
    | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'Exec=libreofficedev7.4' doesn't match anything in file '/nix/store/d1s2babi48pf8iwp2xgc28samzibnc93-libreoffice-7.4.2.3/share/applications/calc.desktop'
Near pkgs/applications/office/libreoffice/default.nix:578:

```
    |
578 |     description = "Comprehensive, professional-quality productivity suite, a variant of openoffice.org";
    | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'Exec=libreofficedev7.4' doesn't match anything in file '/nix/store/d1s2babi48pf8iwp2xgc28samzibnc93-libreoffice-7.4.2.3/share/applications/draw.desktop'
Near pkgs/applications/office/libreoffice/default.nix:578:

```
    |
578 |     description = "Comprehensive, professional-quality productivity suite, a variant of openoffice.org";
    | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'Exec=libreofficedev7.4' doesn't match anything in file '/nix/store/d1s2babi48pf8iwp2xgc28samzibnc93-libreoffice-7.4.2.3/share/applications/impress.desktop'
Near pkgs/applications/office/libreoffice/default.nix:578:

```
    |
578 |     description = "Comprehensive, professional-quality productivity suite, a variant of openoffice.org";
    | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'Exec=libreofficedev7.4' doesn't match anything in file '/nix/store/d1s2babi48pf8iwp2xgc28samzibnc93-libreoffice-7.4.2.3/share/applications/math.desktop'
Near pkgs/applications/office/libreoffice/default.nix:578:

```
    |
578 |     description = "Comprehensive, professional-quality productivity suite, a variant of openoffice.org";
    | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'Exec=libreofficedev7.4' doesn't match anything in file '/nix/store/d1s2babi48pf8iwp2xgc28samzibnc93-libreoffice-7.4.2.3/share/applications/startcenter.desktop'
Near pkgs/applications/office/libreoffice/default.nix:578:

```
    |
578 |     description = "Comprehensive, professional-quality productivity suite, a variant of openoffice.org";
    | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'Exec=libreofficedev7.4' doesn't match anything in file '/nix/store/d1s2babi48pf8iwp2xgc28samzibnc93-libreoffice-7.4.2.3/share/applications/writer.desktop'
Near pkgs/applications/office/libreoffice/default.nix:578:

```
    |
578 |     description = "Comprehensive, professional-quality productivity suite, a variant of openoffice.org";
    | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'Exec=libreofficedev7.4' doesn't match anything in file '/nix/store/d1s2babi48pf8iwp2xgc28samzibnc93-libreoffice-7.4.2.3/share/applications/xsltfilter.desktop'
Near pkgs/applications/office/libreoffice/default.nix:578:

```
    |
578 |     description = "Comprehensive, professional-quality productivity suite, a variant of openoffice.org";
    | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: unused-argument
Unused argument: `commonsLogging`.
Near pkgs/applications/office/libreoffice/default.nix:89:3:

```
   |
89 | , commonsLogging
   |   ^
```
</details>
libreoffice:
<details>

got build log for '/nix/store/mrsqm6fqxgnz0wp8qp9a1flhfjkxjqbh-libreoffice-7.3.6.2' from 'daemon'
When evaluating attribute ‘libreoffice’:
warning: unused-argument
Unused argument: `commonsLogging`.
Near pkgs/applications/office/libreoffice/default.nix:89:3:

```
   |
89 | , commonsLogging
   |   ^
```
</details>
libreoffice-qt:
<details>

got build log for '/nix/store/qc2830hqflb75g5c62psjqf4p3hyvxxq-libreoffice-7.4.2.3' from 'daemon'
When evaluating attribute ‘libreoffice-qt’:
warning: unused-argument
Unused argument: `commonsLogging`.
Near pkgs/applications/office/libreoffice/default.nix:89:3:

```
   |
89 | , commonsLogging
   |   ^
```
</details>
libreoffice-unwrapped:
<details>

got build log for '/nix/store/g11zwnzlyssi8qi036s4v8z1szmsbcja-libreoffice-7.3.6.2' from 'daemon'
When evaluating attribute ‘libreoffice-unwrapped’:
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'Exec=libreofficedev7.3' doesn't match anything in file '/nix/store/g11zwnzlyssi8qi036s4v8z1szmsbcja-libreoffice-7.3.6.2/share/applications/base.desktop'
Near pkgs/applications/office/libreoffice/default.nix:578:

```
    |
578 |     description = "Comprehensive, professional-quality productivity suite, a variant of openoffice.org";
    | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'Exec=libreofficedev7.3' doesn't match anything in file '/nix/store/g11zwnzlyssi8qi036s4v8z1szmsbcja-libreoffice-7.3.6.2/share/applications/calc.desktop'
Near pkgs/applications/office/libreoffice/default.nix:578:

```
    |
578 |     description = "Comprehensive, professional-quality productivity suite, a variant of openoffice.org";
    | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'Exec=libreofficedev7.3' doesn't match anything in file '/nix/store/g11zwnzlyssi8qi036s4v8z1szmsbcja-libreoffice-7.3.6.2/share/applications/draw.desktop'
Near pkgs/applications/office/libreoffice/default.nix:578:

```
    |
578 |     description = "Comprehensive, professional-quality productivity suite, a variant of openoffice.org";
    | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'Exec=libreofficedev7.3' doesn't match anything in file '/nix/store/g11zwnzlyssi8qi036s4v8z1szmsbcja-libreoffice-7.3.6.2/share/applications/impress.desktop'
Near pkgs/applications/office/libreoffice/default.nix:578:

```
    |
578 |     description = "Comprehensive, professional-quality productivity suite, a variant of openoffice.org";
    | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'Exec=libreofficedev7.3' doesn't match anything in file '/nix/store/g11zwnzlyssi8qi036s4v8z1szmsbcja-libreoffice-7.3.6.2/share/applications/math.desktop'
Near pkgs/applications/office/libreoffice/default.nix:578:

```
    |
578 |     description = "Comprehensive, professional-quality productivity suite, a variant of openoffice.org";
    | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'Exec=libreofficedev7.3' doesn't match anything in file '/nix/store/g11zwnzlyssi8qi036s4v8z1szmsbcja-libreoffice-7.3.6.2/share/applications/startcenter.desktop'
Near pkgs/applications/office/libreoffice/default.nix:578:

```
    |
578 |     description = "Comprehensive, professional-quality productivity suite, a variant of openoffice.org";
    | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'Exec=libreofficedev7.3' doesn't match anything in file '/nix/store/g11zwnzlyssi8qi036s4v8z1szmsbcja-libreoffice-7.3.6.2/share/applications/writer.desktop'
Near pkgs/applications/office/libreoffice/default.nix:578:

```
    |
578 |     description = "Comprehensive, professional-quality productivity suite, a variant of openoffice.org";
    | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: stale-substitute
Stale substituteInPlace detected.
substituteStream(): WARNING: pattern 'Exec=libreofficedev7.3' doesn't match anything in file '/nix/store/g11zwnzlyssi8qi036s4v8z1szmsbcja-libreoffice-7.3.6.2/share/applications/xsltfilter.desktop'
Near pkgs/applications/office/libreoffice/default.nix:578:

```
    |
578 |     description = "Comprehensive, professional-quality productivity suite, a variant of openoffice.org";
    | ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md
warning: unused-argument
Unused argument: `commonsLogging`.
Near pkgs/applications/office/libreoffice/default.nix:89:3:

```
   |
89 | , commonsLogging
   |   ^
```
</details>
unoconv:
<details>

got build log for '/nix/store/7plfxdaf40a7pxbcqcc00fzfifk5cb7n-unoconv-0.9.0' from 'daemon'
When evaluating attribute ‘unoconv’:
warning: unclear-gpl
`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/text/unoconv/default.nix:36:5:

```
   |
36 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
</details>

File: pkgs/applications/office/libreoffice/default.nix [hunk 2]
```diff
@@ -364,6 +364,12 @@ in
...
+        --replace "Exec=libreoffice${major}.${minor}"    "Exec=libreoffice"
```
Comment (gador) [inline pkgs/applications/office/libreoffice/default.nix:370]:
```suggestion
        --replace "Exec=libreoffice${major}.${minor}"    "Exec=$out/bin/ssofice"
```

**Comment by gador** [comment]
as seen above `substituteStream(): WARNING: pattern 'Exec=libreofficedev7.3' doesn't match anything in file` the `dev` version should probably be removed.

Also the final link should not be `libreoffice` but the `/nix/store` path

**Comment by ianmjones** [comment]
@gador thanks for the review.

I'm currently building and testing with the last of those automated issues fixed, and will push that up if all looks ok.

**Comment by ianmjones** [comment]
Result of `nixpkgs-review pr 196994` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>jpsxdec</li>
    <li>libreoffice (libreoffice-still)</li>
    <li>libreoffice-fresh</li>
    <li>libreoffice-fresh-unwrapped</li>
    <li>libreoffice-qt</li>
    <li>libreoffice-still-unwrapped</li>
    <li>openlp</li>
    <li>openlpFull</li>
    <li>paperwork</li>
    <li>python310Packages.paperwork-backend</li>
    <li>python310Packages.paperwork-shell</li>
    <li>python39Packages.paperwork-backend</li>
    <li>python39Packages.paperwork-shell</li>
    <li>unoconv</li>
  </ul>
</details>

**Review by gador — APPROVED** [review]
LGTM! Thanks for your PR!

File: pkgs/applications/office/libreoffice/default.nix [hunk 3]
```diff
@@ -364,6 +364,11 @@ in
     ln -s $out/bin/soffice $out/bin/libreoffice
     ln -s $out/lib/libreoffice/share/xdg $out/share/applications
 
+    for f in $out/share/applications/*.desktop; do
+      substituteInPlace "$f" \
+        --replace "Exec=libreoffice${major}.${minor}"    "Exec=$out/bin/soffice"
```
Comment (symphorien) [inline pkgs/applications/office/libreoffice/default.nix:369]: absolute paths in Exec= are problematic, see https://github.com/NixOS/nixpkgs/pull/68035

Comment (gador) [reply pkgs/applications/office/libreoffice/default.nix:369]: wow, that is just another rabbit hole..I guess the easiest thing is to leave it at `"Exec=libreoffice"` then..

Comment (ianmjones) [reply pkgs/applications/office/libreoffice/default.nix]: Thanks @symphorien, good info!

I've put `Exec` back to using `libreoffice`.

**Comment by gador** [comment]
Give me a minute (or an hour..) to re-run `nixpkgs-review`.. :smile:

**Comment by 7c6f434c** [comment]
> Give me a minute (or an hour..) to re-run nixpkgs-review..

Exactly why it should be in the wrapper from the step one, by the way

**Comment by gador** [comment]
Hopefully we do not change the desktop files in the future ;-)

**Comment by 7c6f434c** [comment]
«Hopefully we will not change X in the future» is _never_ a good expectation

**Comment by gador** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 196994` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
14 packages built:
</summary>
<ul>
<li>jpsxdec</li>
<li>libreoffice (libreoffice-still)</li>
<li>libreoffice-fresh</li>
<li>libreoffice-fresh-unwrapped</li>
<li>libreoffice-qt</li>
<li>libreoffice-still-unwrapped</li>
<li>openlp</li>
<li>openlpFull</li>
<li>paperwork</li>
<li>python310Packages.paperwork-backend</li>
<li>python310Packages.paperwork-shell</li>
<li>python39Packages.paperwork-backend</li>
<li>python39Packages.paperwork-shell</li>
<li>unoconv</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
libreoffice-fresh:
<details>

got build log for '/nix/store/2wcpxm5bzm7x5i8z6vp25iw7286f5f5s-libreoffice-7.4.2.3' from 'daemon'
When evaluating attribute ‘libreoffice-fresh’:
warning: unused-argument
Unused argument: `commonsLogging`.
Near pkgs/applications/office/libreoffice/default.nix:89:3:

```
   |
89 | , commonsLogging
   |   ^
```
</details>
libreoffice-fresh-unwrapped:
<details>

got build log for '/nix/store/cskan1qqyv97q7l4i679a0srvbd1dvc9-libreoffice-7.4.2.3' from 'daemon'
When evaluating attribute ‘libreoffice-fresh-unwrapped’:
warning: unused-argument
Unused argument: `commonsLogging`.
Near pkgs/applications/office/libreoffice/default.nix:89:3:

```
   |
89 | , commonsLogging
   |   ^
```
</details>
libreoffice:
<details>

got build log for '/nix/store/2g7vb8hrg9771bvrn13sxbhbqc42dh0m-libreoffice-7.3.6.2' from 'daemon'
When evaluating attribute ‘libreoffice’:
warning: unused-argument
Unused argument: `commonsLogging`.
Near pkgs/applications/office/libreoffice/default.nix:89:3:

```
   |
89 | , commonsLogging
   |   ^
```
</details>
libreoffice-qt:
<details>

got build log for '/nix/store/8970kx56pqwva83vlhbdpyd46fxw26gc-libreoffice-7.4.2.3' from 'daemon'
When evaluating attribute ‘libreoffice-qt’:
warning: unused-argument
Unused argument: `commonsLogging`.
Near pkgs/applications/office/libreoffice/default.nix:89:3:

```
   |
89 | , commonsLogging
   |   ^
```
</details>
libreoffice-still-unwrapped:
<details>

got build log for '/nix/store/50flyy9656nldddywd8k7v9j2px4451m-libreoffice-7.3.6.2' from 'daemon'
When evaluating attribute ‘libreoffice-still-unwrapped’:
warning: unused-argument
Unused argument: `commonsLogging`.
Near pkgs/applications/office/libreoffice/default.nix:89:3:

```
   |
89 | , commonsLogging
   |   ^
```
</details>
unoconv:
<details>

got build log for '/nix/store/z1rm1k3cqw36kpn9h4nnhnh9kgwvwnkb-unoconv-0.9.0' from 'daemon'
When evaluating attribute ‘unoconv’:
warning: unclear-gpl
`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/text/unoconv/default.nix:36:5:

```
   |
36 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
</details>

**Comment by ianmjones** [comment]
Result of `nixpkgs-review pr 196994` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages built:</summary>
  <ul>
    <li>jpsxdec</li>
    <li>libreoffice (libreoffice-still)</li>
    <li>libreoffice-fresh</li>
    <li>libreoffice-fresh-unwrapped</li>
    <li>libreoffice-qt</li>
    <li>libreoffice-unwrapped (libreoffice-still-unwrapped)</li>
    <li>paperwork</li>
    <li>python310Packages.paperwork-backend</li>
    <li>python310Packages.paperwork-shell</li>
    <li>python39Packages.paperwork-backend</li>
    <li>python39Packages.paperwork-shell</li>
    <li>unoconv</li>
  </ul>
</details>

**Pull request merged**